include/godot_cpp/core/class_db.hpp

```diff
@@ -203,6 +203,7 @@ class ClassDB {
 template <typename T, bool is_abstract>
 void ClassDB::_register_class(bool p_virtual, bool p_exposed, bool p_runtime) {
 	static_assert(TypesAreSame<typename T::self_type, T>::value, "Class not declared properly, please use GDCLASS.");
+	static_assert(!std::is_abstract_v<T> || is_abstract, "Class is abstract, please use GDREGISTER_ABSTRACT_CLASS.");
```

Comment:
- if this class is real a no-abstract class, not check, so it can register by `GDREGISTER_CLASS` `GDREGISTER_VIRTUAL_CLASS` `GDREGISTER_ABSTRACT_CLASS` `GDREGISTER_INTERNAL_CLASS` `GDREGISTER_RUNTIME_CLASS` etc.. (all register methods)
- and if this class is real an abstract class, check `is_abstract` param, make this class must register by `GDREGISTER_ABSTRACT_CLASS`